### ISSUE TITLE
Measurement unit conversion

### ISSUE THREAD

**Issue by sffc** [unicode-org/icu4x#2196]
We need to implement measurement unit conversion in order to support full-stack localized measurement formatting.

This is not in and of itself the job of an i18n library, so I think it should go into utilities. I see this as similar to the TZIF calculations for time zones, which are also not themselves i18n but which we need for other i18n components.

This is an advanced issue because it requires a lot of API design work.

- Part of #2192

**Comment by zbraniecki** [comment]
I agree that it should be a `util/` and if possible, we should use an existing utility like https://crates.io/crates/unit-conversions
